### PR TITLE
Simpler implementation of case-insensitive string comparisons

### PR DESCRIPTION
**Pull request by marcomagdy**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by justinboswell** [comment]
This is visually simpler, but for something that is going to be called very heavily in our HTTP parsing, 8 ops per character ([sub, lt, branch, or] for each character in the compared arrays) vs a small table compare is going to lose every time on runtime speed.

**Comment by marcomagdy** [comment]
> This is visually simpler, but for something that is going to be called very heavily in our HTTP parsing, 8 ops per character ([sub, lt, branch, or] for each character in the compared arrays) vs a small table compare is going to lose every time on runtime speed.

Not true. We benchmarked it with google's microbenchmark framework. They're practically the same.
But anyways. I just thought it's more elegant

File: source/byte_buf.c [hunk 1]
```diff
@@ -227,20 +227,15 @@ bool aws_byte_cursor_eq(const struct aws_byte_cursor *a, const struct aws_byte_c
...
-    198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219,
-    220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241,
-    242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255};
+static int isupper(uint8_t ch) {
+    return ch - 'A' < 26;
+}
+
+static uint8_t tolower(uint8_t ch) {
+    if (isupper(ch))
+        return ch | 32;
```
[the remark this answers is not in the document]
Comment (marcomagdy) [reply source/byte_buf.c]: No. Not minus 32.

[the remark this answers is not in the document]
Comment (singku) [reply source/byte_buf.c]: Yeah, I read it incorrectly. It should be plus 32, but `| 32` is a safe way to achieve that.

**Pull request closed without merging**